scripts/Install-Extension.ps1

```diff
@@ -0,0 +1,249 @@
+<#
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: MIT
+
+.SYNOPSIS
+This script provides helpers to install or uninstall ebpf extensions.
+
+.PARAMETER Extension
+    Specifies the extension to install. This MUST be one of ("neteventebpfext", "ntosebpfext").
+
+.PARAMETER Action
+    Specifies the action to take. This MUST be either "Install" or "Uninstall".
+
+.PARAMETER BinaryDirectory
+    Specifies the directory containing the necessary binaries().
+    This MUST be the full path to the directory. By default, the current directory is used.
+
+.EXAMPLE
+    Install_Extension.ps1 -Action Install
+
+.EXAMPLE
+    Install_Extension.ps1 -Action Uninstall -Extension "neteventebpfext" 
+
+.EXAMPLE
+    Install_Extension.ps1 -Action Install -BinaryDirectory "C:\binaries"
+#>
+
+param (
+    [Parameter(Mandatory=$true)]
+    [ValidateSet("Install", "Uninstall")]
+    [string]$Action,
+    [Parameter(Mandatory=$true)]
+    [ValidateSet("neteventebpfext", "ntosebpfext")]
+    [string]$Extension,
+    [Parameter(Mandatory=$false)]
+    [string]$BinaryDirectory = (Get-Location).Path
+)
+
+Set-StrictMode -Version 'Latest'
+$PSDefaultParameterValues['*:ErrorAction'] = 'Stop'
+
+<#
+.SYNOPSIS
+Stops and deletes a service.
+
+.PARAMETER ServiceName
+The name of the service to cleanup.
+
+.EXAMPLE
+Clear-Service -ServiceName "neteventebpfext"
+#>
+function Clear-Service(
+    [Parameter(Mandatory=$true)]
+    [string]$ServiceName
+) {
+    # Wait for the service to stop.
+    $StopSuccess = $false
+    try {
+        Stop-Service $ServiceName
+        for ($i = 0; $i -lt 100; $i++) {
+            if (-not (Get-Service $ServiceName -ErrorAction Ignore) -or
+                (Get-Service $ServiceName).Status -eq "Stopped") {
+                $StopSuccess = $true
+                break;
+            }
+            Start-Sleep -Milliseconds 100
+        }
+        if (!$StopSuccess) {
+            Write-Verbose "$ServiceName failed to stop"
+        }
+    } catch {
+        Write-Verbose "Exception while waiting for $ServiceName to stop"
+    }
+
+    # Delete the service.
+    if (Get-Service $ServiceName -ErrorAction Ignore) {
+        try { sc.exe delete $ServiceName > $null }
+        catch { Write-Verbose "'sc.exe delete $ServiceName' threw exception!" }
+
+        # Wait for the service to be deleted.
+        $DeleteSuccess = $false
+        for ($i = 0; $i -lt 10; $i++) {
```

Comment:
Is 10 retries 10ms apart long enough for the service to be consistently deleted?